### PR TITLE
[SPARK-36314][SS] Update Sessionization examples to use native support of session window

### PR DESCRIPTION
**Pull request by HeartSaVioR**
### What changes were proposed in this pull request?

This PR proposes to update Sessionization examples to use native support of session window. It also adds the example for PySpark as native support of session window is available to PySpark as well.

### Why are the changes needed?

We should guide the simplest way to achieve the same workload. I'll provide another example for cases we can't do with native support of session window.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manually tested.

### CODE REVIEW

**Comment by HeartSaVioR** [comment]
Target branch is master/3.2.

**Comment by HeartSaVioR** [comment]
cc. @viirya @xuanyuanking

File: examples/src/main/python/sql/streaming/structured_sessionization.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import sys
+
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import explode
+from pyspark.sql.functions import split
+from pyspark.sql.functions import count, session_window
+
+if __name__ == "__main__":
+    if len(sys.argv) != 3 and len(sys.argv) != 2:
+        msg = "Usage: structured_network_wordcount_windowed.py <hostname> <port> "
```
Comment (viirya) [inline examples/src/main/python/sql/streaming/structured_sessionization.py:42]: structured_sessionization.py?

**Review by viirya — COMMENTED** [review]
Looks good. One typo found.

**Review by xuanyuanking — APPROVED** [review]
LGTM

**Review by viirya — APPROVED** [review]

**Comment by viirya** [comment]
Thanks! Merging to master/3.2.

**Comment by HeartSaVioR** [comment]
Thanks all for the quick reviews and merge!

**Comment by thenooz** [comment]
Hello fellow spark structured streaming sessionizers. According to the apache spark docs, output mode update is not allowed with session window. I tried running the example code in my local machine for 3.2.1 and 3.2.0 and it failed with error: 

> org.apache.spark.sql.AnalysisException: Update output mode not supported for session window on streaming DataFrames/DataSets;

According to this example code it can be done. What am I missing?

**Comment by HeartSaVioR** [comment]
Ah we seem to miss updating the example... We allowed update mode first and disallowed due to unclear semantic.

Could you please raise a PR to update the output mode to append for Scala/Java/Python examples for session window if you don't mind? I can jump in if you're not available to address this. Thanks in advance!

**Comment by thenooz** [comment]
Sure, but you see, if we do it with append mode, then that means the sessions will be returned only after they are closed. Due to this, possibility and purpose of real-time session tagging is defeated. Should we not revert the example to **mapGroupsWithState** method?

**Comment by HeartSaVioR** [comment]
We still have an example of session window against (flat)mapGroupsWithState. We just don't want to duplicate the code example.

The main reason I proposed dropping the functionality of update mode for session window was due to the characteristic of session window. When someone uses update mode for the streaming query, it does not only mean they want to get update immediately, but also mean they are going to "upsert" the target table.

Upsertion is normally performed by replacing the existing value(s) for specific key to new value(s). If we consider the out-of-order events, session window will expand arbitrarily - both start and end of session can change. (Technically saying, the old example of session window was incorrect in terms of event-time semantic.)  Given Spark does not produce the old value(s), the query can't always safely upsert the result. E.g. user A has multiple sessions in the target table and some of sessions were already closed. Spark never gives them, and the query is in a risk of dropping these old-but-valid sessions.

**Comment by thenooz** [comment]
Got it. So the problem is we run a risk of taking wrong real-time decisions on the basis of immediate updates and it is very much possible the session we calculated was incorrect due to late events. So, the safer and more accurate solution would be append mode with watermarking.
But still, back to a probably annoying question: Can we allow update mode with session window with a warning? You know, for people who take big risks.
Also, will check on updating the example code. I am new to this.
Thank you very much.

**Pull request closed without merging**